test_scripts/API/ButtonSubscription/common_buttonSubscription.lua

```diff
@@ -0,0 +1,308 @@
+  EXPECT_HMICALL("Buttons." .. pRpc,{ appID = m.getHMIAppId(pAppId), buttonName = pButtonName })
+    :Times(0)
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = pResultCode })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+    :Times(0)
+end
+
+function m.buttonPress(pAppId, pButtonName)
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONDOWN", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonPress",
+    { name = pButtonName, mode = "SHORT", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONUP", appID = m.getHMIAppId(pAppId) })
+  m.getMobileSession(pAppId):ExpectNotification( "OnButtonEvent",
+    { buttonName = pButtonName, buttonEventMode = "BUTTONDOWN"},
+    { buttonName = pButtonName, buttonEventMode = "BUTTONUP"})
+    :Times(2)
+  m.getMobileSession(pAppId):ExpectNotification( "OnButtonPress",
+    { buttonName = pButtonName, buttonPressMode = "SHORT"})
+end
+
+function m.buttonPress_CUSTOM_BUTTON(pButtonName, pCustomButtonID, pAppID)
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent", 
+      { name = pButtonName, mode = "BUTTONDOWN", customButtonID = pCustomButtonID, appID = m.getHMIAppId(pAppID) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonPress", 
+      { name = pButtonName, mode = "SHORT", customButtonID = pCustomButtonID, appID = m.getHMIAppId(pAppID) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent", 
+      { name = pButtonName, mode = "BUTTONUP", customButtonID = pCustomButtonID, appID = m.getHMIAppId(pAppID) })
+  m.getMobileSession(pAppID):ExpectNotification("OnButtonEvent", 
+      { buttonName = pButtonName, buttonEventMode = "BUTTONDOWN", customButtonID = pCustomButtonID },
+      { buttonName = pButtonName, buttonEventMode = "BUTTONUP", customButtonID = pCustomButtonID })
+  :Times(2)
+  m.getMobileSession(pAppID):ExpectNotification("OnButtonPress", 
+      { buttonName = pButtonName, buttonPressMode = "SHORT", customButtonID = pCustomButtonID })
+end 
+
+function m.buttonPressUnsuccess(pAppId, pButtonName)
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONDOWN", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonPress",
+    { name = pButtonName, mode = "SHORT", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONUP", appID = m.getHMIAppId(pAppId) })
+  m.getMobileSession(pAppId):ExpectNotification( "OnButtonEvent")
+    :Times(0)
+  m.getMobileSession(pAppId):ExpectNotification("OnButtonPress")
+    :Times(0)
+end
+
+function m.buttonPressUnsuccess_CUSTOM_BUTTON(pButtonName, pCustomButtonID, pAppID)
```

Comment:
There is no need to define buttonPressUnsuccess_CUSTOM_BUTTON, add parameter pCustomButtonID to buttonPressUnsuccess, in case pCustomButtonID will be empty the OnButtonEvent and OnButtonPress wiil be sent as without custom id